assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -14530,9 +14538,22 @@ else if (deleted.equalsIgnoreCase(Boolean.TRUE.toString()) && (a.getContent().ge
 			}
 			else
 			{
-				// read all Assignments
-				returnResources = assignmentService.getListAssignmentsForContext((String) state
-						.getAttribute(STATE_CONTEXT_STRING));
+				if (isSynopticTool() & isOnWorkspaceTab()) {
+					//Get all sites for this user
+					String userId = sessionManager.getCurrentSessionUserId();
+	                List<Site> sites = siteService.getSites(SelectionType.ACCESS, null, null, null,null, null);
+	                for (Site site : sites) {
+	                	Member sm = site.getMember(userId);
+	                	if (sm != null) {
+	                		returnResources.addAll(assignmentService.getListAssignmentsForContext(site.getId()));
```

Comment:
While this works might be worth pushing this up into the API so that you can pass in a list of `siteIds` to get assignments for, rather than going back to the DB for each one.

Also without paging could this end up causing problems where archiving of old course sites doesn't happen quickly and a tutor can end up with 100s of assignments being retrieved.

Reply:
Seen any production issues from this at all?

Reply:
No we never did, but only ran it on a few clients that probably didn't have a lot of sites. It looks like there was a method added here by SAK-23358 (getMyAssignmentsForAllSite) that does some of this already and a few other tools do this too. I feel like any large site enrollment could cause an issue. Maybe it should just be more limited.